src/_P101_WakeOnLan.ino

```diff
@@ -0,0 +1,475 @@
+          msgStr  = wolStr;
+          msgStr += F("MAC= ");
+          msgStr += paramMac;
+          msgStr += F(", IP= ");
+          msgStr += paramIp;
+          msgStr += F(", Port= ");
+          msgStr += paramPort;
+          addLog(LOG_LEVEL_INFO, msgStr);
+
+          // Send Magic Packet.
+          if (WiFi.status() == WL_CONNECTED) {
+            IPAddress local_IP;
+            local_IP.fromString(paramIp);
+            WOL.setBroadcastAddress(local_IP);
+
+            // WOL.setRepeat(1, 0); // One Magic Packet, No Repeats. (Library default)
+
+            if (!WOL.sendMagicPacket(paramMac, paramPort.toInt())) {
+              msgStr  = wolStr;
+              msgStr += F("Error, Magic Packet Failed (check parameters)");
+              addLog(LOG_LEVEL_INFO, msgStr);
+            }
+          }
+          else {
+            msgStr  = wolStr;
+            msgStr += F("Error, WiFi Off-Line");
+            addLog(LOG_LEVEL_INFO, msgStr);
+          }
+        }
+      }
+      break;
+    }
+
+    case PLUGIN_EXIT:
+    {
+      break;
+    }
+  }
+  return success;
+}
+
+// ************************************************************************************************
+// validateIp(): IPv4 Address validity checker.
+// Arg: IP Address String in dot separated format (192.168.1.255).
+// Return true if IP string appears legit.
+bool validateIp(String ipStr) {
+  IPAddress ip;
+  unsigned int length = ipStr.length();
+
+  if ((length < IP_MIN_SIZE_P101) || (length > IP_ADDR_SIZE_P101)) {
+    return false;
+  }
+  else if (ip.fromString(ipStr) == false) { // ThomasTech's Trick to Check IP for valid formatting.
+    return false;
+  }
+
+  return true;
+}
+
+// ************************************************************************************************
+// validateMac(): MAC Address validity checker.
+// Arg: MAC Address String in colon separated format (00:00:00:00:00:00).
+// Return true if MAC string appears legit.
+bool validateMac(String macStr) {
+  uint8_t pos = 0;
+  char    hexChar;
+
+  if (macStr.length() != MAC_ADDR_SIZE_P101) {
+    return false;
+  }
+
+  for (uint8_t strPos = 0; strPos < MAC_ADDR_SIZE_P101; strPos++) {
+    uint8_t mod = strPos % 3;
+
+    if (mod == 2) {
+      if (macStr[strPos] == MAC_SEP_CHAR_P101) { // Must be a colon in the third position.
+        pos++;
+      }
+      else {
+        return false;
+      }
+    }
+    else {
+      hexChar = macStr[strPos];
+
+      if (!isHexadecimalDigit(hexChar)) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+// ************************************************************************************************
+// validatePort(): Simple UDP Port Number validity checker.
+// Arg: Port Number String (decimal).
+// Return true if Port string appears legit.
+bool validatePort(String portStr) {
+  bool pass = true;
+  long portNumber;
+
+  portNumber = portStr.toInt();
+
+  if ((portNumber < 0) || (portNumber > PORT_MAX_P101)) {
```

Comment:
Is portnr 0 a valid port?
That's the return value of `toInt()` when it is a string with non-numerical characters.

Reply:
Port 0 is valid. Regarding non-numerics, there's a very simple isDigit() test at line 466.  This only checks the leading char, which is enough to find blatant port value mistakes in the command sentence.